### PR TITLE
Add more M3 polymers

### PR DESCRIPTION
**Pull request by fgrunewald**
Martini3 model for PMMA, PMA, PE, and small update on PS. PMA is verified against Rg data in butanone, amorphous density, solubility trends in water, DMSO, Butanone, Acetone. PE comes from the LTF and is further unverified. 

Polymers implemented:
- [x] PEO
- [x] PMA
- [x] PMMA
- [x] PE
- [x] PS
- [x] P3HT
- [x] PVA
- [x] PSS
- [x] Dextran

### CODE REVIEW

**Comment by fgrunewald** [comment]
@pckroon Do you perhaps have an idea why commit with tag a886e47 failed but changing the angles to the format as in commit c268c5c? Commit a886e47 works on my lab-machine and on the cluster but fails on the CI and my notebook at home. This is something that is potentially worrisome unless I'm overlooking a simple mistake.

**Comment by pckroon** [comment]
The only thing I can come up with is the lack of empty newline at the end of the file. That may (or may not) affect the parsing of the conditional. Otherwise it may be a matter of the conditional not being interpreted at all?

**Comment by ricalessandri** [comment]
In case like this one:
https://github.com/marrink-lab/polyply_1.0/blob/eb5602b95991ef442a0c8c72bc1290ab2bcd2e86/polyply/tests/test_data/library_tests/martini3/PS/polyply/PS.itp#L34-L88

you can do with fewer lines; namely:
```
[ bonds ]
 1  4 1 0.27 8000 
...
20 21 1 0.27 8000
#ifndef FLEXIBLE
[ constraints ]
#endif
 2  3 1 0.290 10000
...
 21 22 1 0.290 10000
```
because the force constants are not read by GROMACS in the `[ constraints ]` section.

**Comment by fgrunewald** [comment]
Thanks! I actually did not know this. But I am 100% certain polyply cannot parse that syntax. So we'd be shooting ourselves in the foot.

**Comment by fgrunewald** [comment]
@pckroon I actually found a bug in the library test script that would cause it to not fail, when an interaction is there but has different parameters. I fixed it, I believe. Can one add a controlled fail or something? Also I added the meta values to the comparison, so that it checks ifdef statements.

File: polyply/tests/test_lib_files.py [hunk 1]
```diff
@@ -85,7 +85,8 @@ def assert_equal_blocks(block1, block2):
                     if _interaction_equal(ref_inter, new_inter, inter_type):
                         break
                 else:
-                    return False
+                    assert False
+        assert True
```
Comment (pckroon) [inline polyply/tests/test_lib_files.py:89]: This is a void statement and should be removed

File: polyply/tests/test_lib_files.py [hunk 2]
```diff
@@ -85,7 +85,8 @@ def assert_equal_blocks(block1, block2):
                     if _interaction_equal(ref_inter, new_inter, inter_type):
                         break
                 else:
-                    return False
+                    assert False
```
Comment (pckroon) [inline polyply/tests/test_lib_files.py:88]: Consider `pytest.fail()`: https://docs.pytest.org/en/6.2.x/reference.html#pytest-fail
And give the reason

File: polyply/tests/test_lib_files.py [hunk 3]
```diff
@@ -85,7 +85,8 @@ def assert_equal_blocks(block1, block2):
                     if _interaction_equal(ref_inter, new_inter, inter_type):
                         break
```
Comment (pckroon) [inline polyply/tests/test_lib_files.py:86]: Doesn't this test just the first interaction?

**Review by pckroon — CHANGES_REQUESTED** [review]

File: polyply/tests/test_lib_files.py (hunk 1 above)
Comment (fgrunewald) [reply polyply/tests/test_lib_files.py:89]: indeed works without ;)

File: polyply/tests/test_lib_files.py (hunk 3 above)
Comment (fgrunewald) [reply polyply/tests/test_lib_files.py:86]: yes it scans new interactions until it finds one match in ref_interactions. This complicated syntax has to do with multiple dihedrals and these kind of things.

**Pull request merged**